docs/chat_message_history.ipynb

```diff
@@ -187,6 +187,30 @@
     ")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AlloyDBEngine for AlloyDB Omni\n",
+    "To create an `AlloyDBEngine` for AlloyDB Omni, you will need to first create an engine with any PostgreSQL-compatible client or code library, and then turn it into an `AlloyDBEngine` by calling `AlloyDBEngine.from_engine()`. Here is an example connection using `SQLAlchemy` with the `Asyncpg` dialect:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sqlalchemy.ext.asyncio import create_async_engine\n",
+    "\n",
+    "# Replace with your own AlloyDB Omni info\n",
+    "conn_string = (\n",
+    "    f\"postgresql+asyncpg://{OMNI_USER}:{OMNI_PASSWORD}@{OMNI_HOST}:5432/{OMNI_DATABASE}\"\n",
+    ")\n",
+    "async_engine = sqlalchemy.ext.asyncio.create_async_engine(conn_string)\n",
+    "engine = AlloyDBEngine.from_engine(async_engine)"
```

Comment:
This only currently works with async methods correct?

Reply:
I was able to get the async engine working. But followed https://github.com/googleapis/langchain-google-alloydb-pg-python/pull/180 to pass the loop param here: https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/7e3a5ee29d9ec2e912e6ab708a366c7a0744a52e/src/langchain_google_alloydb_pg/engine.py#L315-L318
Like this:
```
        loop = asyncio.new_event_loop()
        thread = Thread(target=loop.run_forever, daemon=True)
        thread.start()
        return cls(cls.__create_key, engine, loop, None)
```

Reply:
The setup seems to be same for [here](https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/ead243e9173175cfbe923547a0aaa4ff19055391/docs/vector_store.ipynb#L257) as well.

Reply:
The sync engine will throw [here](https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/ead243e9173175cfbe923547a0aaa4ff19055391/src/langchain_google_alloydb_pg/engine.py#L327-L328) complaining:

`TypeError: 'Connection' object does not support the asynchronous context manager protocol`.

```
import sqlalchemy
from google.cloud.alloydb.connector import Connector

connector = Connector(refresh_strategy="background")
sync_engine = sqlalchemy.create_engine(
    sqlalchemy.engine.url.URL.create(
        drivername="postgresql+pg8000",
        username=OMNI_USER,
        password=OMNI_PASSWORD,
        host=OMNI_HOST,
        port=OMNI_PORT,  # Default PostgreSQL port = 5432
        database=OMNI_DATABASE,
    ),
)

```

Reply:
@vishwarajanand `sqlalchemy.create_engine` creates a synchronous `Engine` while right now our package only supports async engines as `from_engine` expects an `AsyncEngine`. These two classes have different interfaces which is the why you get the error you are seeing.

Reply:
@jackwotherspoon I am with you on that sync engine is not yet supported. Just that async engine also needed a code change.

Let me know if there's more changes needed here.

Reply:
Uploaded to latest code